src/init.cpp

```diff
@@ -1226,6 +1226,136 @@ bool AppInitLockDataDirectory()
     return true;
 }
 
+static bool AppInitLoadBlockIndex(const CChainParams& chainparams, int64_t nCoinDBCache, int64_t nBlockTreeDBCache, bool fReindexChainState, bool& fReset, std::string& strLoadError)
+{
+    try {
```

Comment:
I really like this refactor, but I wanted to comment briefly on the `try` block.

I suspect that you kept the `try` block to keep things simple, but alternatively you can also use a _function try-block_ if you like the notational convenience that it provides. 

A function try-block will save one level of indentation in the function body, which could be nice with a function body of this size.

Function try-blocks are not commonly used,  but it is perfectly fine to use them (at least according to Stroustrup). 

With a function-try block this function would look something like this:

```c++
static bool AppInitLockBlockIndex(...) try 
{
    ...
    return true;
} catch (std::exception& e) 
{
    ...
    return false;
}
```

Of course this comment is not critical, feel free to ignore it.

Reply:
Wow, TIL. Read up on them and I'm into.

Reply:
Switched back as per https://github.com/bitcoin/bitcoin/pull/13582#issuecomment-403509243